usl_models/usl_models/flood_ml/batch_predictor.py

```diff
@@ -0,0 +1,191 @@
+        )
+
+    def set_model_run_metadata(self, scenario_ids: list[str]):
+        """Sets model run metadata in firestore."""
+        model_ref = self.db.collection("models").document(self.model_id)
+        run_ref = model_ref.collection("runs").document(self.run_id)
+        run_ref.set({"scenario_ids": scenario_ids})
+
+    def set_model_run_prediction_metadata(
+        self,
+        study_area_id: str,
+        scenario_id: str,
+        chunk_metadata: list[dict],
+    ):
+        """Sets the model run prediction metadata in firestore."""
+        model_ref = self.db.collection("models").document(self.model_id)
+        run_ref = model_ref.collection("runs").document(self.run_id)
+        prediction_id = f"Prediction-{study_area_id}-{scenario_id}"
+        prediction_ref = run_ref.collection("predictions").document(prediction_id)
+        prediction_ref.set(
+            {
+                "study_area_id": study_area_id,
+                "scenario_configuration_id": scenario_id,
+                "chunks": chunk_metadata,
+            }
+        )
+
+    def get_chunk_metadata(
+        self, study_area_id: str, config_id: str, chunk_ids: list[str]
+    ) -> list[dict]:
+        """Returns a list of chunk metadata."""
+        chunk_metadata = []
+        for chunk_id in chunk_ids:
+            path = self.get_prediction_npy_path(
+                study_area_id=study_area_id, config_id=config_id, chunk_id=chunk_id
+            )
+            chunk_metadata.append(
+                {
+                    "id": chunk_id,
+                    "path": f"gs://{self.output_bucket}/{path}",
+                }
+            )
+        return chunk_metadata
+
+    @staticmethod
+    def prediction_to_json(prediction: np.ndarray, chunk_id: str) -> dict:
+        """Converts a prediction array to JSON."""
+        return {
+            "instance": {
+                "values": [1],
+                "key": chunk_id,
+            },
+            "prediction": prediction.tolist(),
+        }
+
+    def bundle_predictions_to_jsonl(
+        self,
+        study_area_id: str,
+        config_id: str,
+        chunk_ids: list[str],
+        scenario_id: str,
+    ):
+        """Convert npy files to a single jsonl file."""
+        jsonl_path = (
+            f"{self.run_id}/flood/{self.model_id}/{study_area_id}"
+            + f"/{scenario_id}/prediction.results-1-of-1"
+        )
+        with self.client.bucket(self.output_bucket).blob(jsonl_path).open("w") as fd:
+            for chunk_id in chunk_ids:
+                path = self.get_prediction_npy_path(study_area_id, config_id, chunk_id)
+                logging.info(f"Bundling {path}...")
+                prediction = self.load_prediction_npy(
+                    study_area_id, config_id, chunk_id
+                )
+                json_data = self.prediction_to_json(prediction, chunk_id)
+                json.dump(json_data, fd)
+                fd.write("\n")
```

Comment:
Is there a reason to keep the npy files around past this point? Can we delete them now?

Reply:
Sure, I've added delete logic here.